tests/python/frontend/pytorch/test_forward.py

```diff
@@ -0,0 +1,766 @@
+                name = "input" + str(i)
+                arr = np.random.random(shape).astype("float32")
+                input_data[name] = tvm.nd.array(arr)
+            t_start = time()
+            model.set_input(**input_data)
+            model.run()
+            for i, shape in enumerate(output_shapes):
+                arr = np.zeros(shape).astype("float32")
+                model.get_output(i, tvm.nd.array(arr))
+            t_end = time()
+        count += 1
+        if count < dryruns:
+            continue
+        latencies.append(t_end - t_start)
+        mean = np.mean(latencies)
+        stdev = np.std(latencies)
+        sample_size = len(latencies)
+        if sample_size > dryruns:
+            lower, upper = confidence_interval(mean, stdev, sample_size)
+            est = (upper + lower) / 2
+            err = (upper - lower) / 2
+            if err < thresh:
+                return est
+
+def verify_model(model_name, input_data=[]):
+    """Assert that the output of a compiled model matches with that of its
+    baseline."""
+    if len(input_data) == 0:
+        baseline_model, baseline_input = load_model(model_name)
+    else:
+        baseline_model = model_name
+        baseline_input = input_data
+    if torch.cuda.is_available():
+        baseline_model = baseline_model.cuda()
+        baseline_input = baseline_input.cuda()
+    with torch.no_grad():
+        baseline_outputs = baseline_model(baseline_input)
+    if isinstance(baseline_outputs, tuple):
+        baseline_outputs = tuple(out.cpu().numpy() for out in baseline_outputs)
+    else:
+        baseline_outputs = (baseline_outputs.float().cpu().numpy(),)
+    output_shapes = [out.shape for out in baseline_outputs]
+    dtype = "float32"
+    input_name = "input0"
+    input_shapes = {input_name: list(baseline_input.shape)}
+    trace = torch.jit.trace(baseline_model, baseline_input).float().eval()
+    if torch.cuda.is_available():
+        trace = trace.cuda()
+    else:
+        trace = trace.cpu()
+
+    mod, params = relay.frontend.from_pytorch(trace, input_shapes)
+    compiled_input = {input_name: tvm.nd.array(baseline_input.cpu().numpy())}
+
+    with relay.build_config(opt_level=3):
+        for target, ctx in ctx_list():
+            relay_graph, relay_lib, relay_params = relay.build(mod, target=target, params=params)
+            relay_model = graph_runtime.create(relay_graph, relay_lib, ctx)
+            relay_model.set_input(**relay_params)
+            relay_model.set_input(**compiled_input)
+            relay_model.run()
+
+            for i, baseline_output in enumerate(baseline_outputs):
+                output_shape = baseline_output.shape
+                compiled_output = relay_model.get_output(
+                    i, tvm.nd.array(np.zeros(output_shape).astype(dtype), ctx)).asnumpy()
+
+                compiled_relay_output = relay_model.get_output(
+                    i, tvm.nd.array(np.zeros(output_shape).astype(dtype), ctx)).asnumpy()
```

Comment:
why have compiled_relay_output here? And `get_output` can be simplified. why not just

```
for i, baseline_output in enumerate(baseline_outputs):
    relay_output = relay_model.get_output(i).asnumpy()

    assert_shapes_match(baseline_output, relay_output)
    tvm.testing.assert_allclose(baseline_output, relay_output,
                                rtol=1e-3, atol=1e-3)
```

Reply:
Ah, this was left-over from comparing nnvm and relay parsing.